sdk/appconfiguration/app-configuration/test/public/utils/testHelpers.ts

```diff
@@ -61,19 +56,19 @@ export function getTokenAuthenticationCredential(): CredsAndEndpoint | undefined
         console.log("Functional tests not running - set client identity variables to activate");
       }
 
-      return undefined;
+      throw new Error("Invalid value for requiredEnvironmentVariables");
```

Comment:
You can remove
```
      if (tokenCredentialsNotPresentWarning) {
        tokenCredentialsNotPresentWarning = true;
        console.log("Functional tests not running - set client identity variables to activate");
      }
```

Reply:
If we don't throw an error, then the `connectionString` still has type `string | undefined` so we cannot cleanup the `this.skip()`

Reply:
I want you to throw.
But remove the console log that's not needed.